File: 1260.py
from collections import deque

def bfs(visited, graph, v):
    queue = deque()
    queue.append(v)
    while queue:
        current = queue.popleft()
        if visited[current]:
            continue
        else:
            print(current, end = " ")
            visited[current] = True
            for adj in graph[current]:
                queue.append(adj)

def dfs(visited, graph, v):
  print(v, end = " ")
  visited[v] = True
  for i in graph[v]:
    if not visited[i]:
      dfs(visited, graph, i)
    

n, m, v = map(int, input().split())
visited1 = [False for i in range(n+1)]
visited2 = [False for i in range(n+1)]
graph = [[] for i in range(n+1)]

for i in range(m):
    v1, v2 = map(int, input().split())
    graph[v1].append(v2)
    graph[v2].append(v1)
    graph[v1].sort()
    graph[v2].sort()

dfs(visited1, graph, v)
print()
bfs(visited2, graph, v)

File: 1260.java
import java.io.*;
import java.util.*;

public class Main {
    
    static int N,M,V;
    static int[] visited;
    static int[][] graph;
    static StringBuilder sb = new StringBuilder();
    static LinkedList<Integer> queue = new LinkedList<>();
    
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        String[] info = br.readLine().split(" ");
        N = Integer.parseInt(info[0]);
        M = Integer.parseInt(info[1]);
        V = Integer.parseInt(info[2]);
        
        graph = new int[N+1][N+1];
        visited = new int[N+1];
        
        for(int i=0; i<M; i++) {
            info = br.readLine().split(" ");
            int start = Integer.parseInt(info[0]);
            int end = Integer.parseInt(info[1]);
            graph[start][end] = graph[end][start] = 1;
        }
        
        dfs(V);
        sb.append("\n");
        visited = new int[N+1];
        bfs(V);
        
        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
    
    static void dfs(int now) {
        visited[now] = 1;
        sb.append(String.valueOf(now) + " ");
        
        for(int i=1; i<N+1; i++) {
            if(graph[now][i] == 1 && visited[i] == 0) {
                dfs(i);
            }
        }
    }
    
    static void bfs(int now) {
        visited[now] = 1;
        sb.append(String.valueOf(now) + " ");
        queue.offer(now);
        
        while(!queue.isEmpty()) {
            int current = queue.poll();
            for(int i=1; i<N+1; i++) {
                if(graph[current][i] == 1 && visited[i] == 0) {
                    visited[i] = 1;
                    queue.offer(i);
                    sb.append(String.valueOf(i) + " ");
                }
            }
        }
    }
}
